### ISSUE TITLE
Support cite on blockquotes and perhaps classes

### ISSUE THREAD

**Issue by vr8hub** [typora/typora-issues#1367]
I know Typora doesn't support HTML fragments (I'd love to hear/read more about why that is; the other markdown editors I use don't have a problem with properly displaying HTML fragments in preview mode, provided it's properly handled in the CSS), so it would be nice if support could be added for some common items.

For example, on many blockquotes, <cite></cite> tags are used for the author of the quote. Markdown has blockquote syntax in the ">", but nothing for the cite. It would be great if Typora could at least support that specific tag in the blockquote so it would be displayed live instead of as html source.
> This is the quote
> <cite>By Some Famous Person</cite>

 Also, explicit classes on already supported items, e.g. p or blockquote. Those are two where it's very common to occasionally need to use a class to do slightly different formatting. Again, other editors can do this with no problem; this is a "keep up" situation. Even if Typora can't/won't support HTML fragments, it would be good for it to expand its known vocabulary.

<p class="someclass">This is some test that should display as paragraph, just like <p>, but it should use the .someclass formatting.</p>
And so on for blockquote and perhaps others.

(Lots of suggestions because I *like* using it. :) )

**Comment by abnerlee** [comment]
If typora supports HTML fragments, which HTML tag will be used?

I read from https://stackoverflow.com/questions/5460356/html5-block-quote-with-author, it seems there's no standard tag to represent "By Some Famous Person"

**Comment by vr8hub** [comment]
Really? That question was asked in 2011. That's 1000 years ago in internet terms. :) But even then, if you scroll down the answers, you'll find that someone quoted the HTML 5.3 spec showing the exact thing I mentioned — a `<cite>` tag inside a `<blockquote>`. So now there _is_ a standard tag to represent "By Some Famous Person". (And there has been for years, the HTML spec is finally getting around to formalizing it.)

For more information, see, e.g., [this](http://html5doctor.com/cite-and-blockquote-reloaded/). Specifically this paragraph:
>Previously in HTML5 it is was not conforming to include citations within blockquote elements. Now it is, as long as the citation content is within a cite or footer element. Citations inside blockquote elements are a common markup pattern (_**data indicates approximately 60% of blockquote elements include citations**_), the change to the HTML spec acknowledges this and provides semantic mechanisms to differentiate quoted content from citations. (emphasis added)

(And note that that was written five years ago. So five years ago, *before* there was a "standard", `<cite>` was already used 60% of the time. Think what it is now.)

And [here's](https://w3c.github.io/html/textlevel-semantics.html#the-cite-element) the relevant link for the HTML spec.

That's the specific issue I raised (supporting `<cite>`, and, preferably, `<footer>` as well, since `<cite>` is normally put in a `<footer>`; that allows the author's name to be right-justified, e.g.).

But to answer your question, the point of support HTML fragments is supporting whatever HTML fragments get used. Again, this is something most of the competition does. And I would think it would make even more sense for Typora, since it appears to be largely HTML-based.

And even if you want to narrow the answer, I already gave you the narrowing — at the least we should be able to put classes on already-supported tags.
